dart/lib/src/sentry.dart

```diff
@@ -1,15 +1,22 @@
 import 'dart:async';
 
+import 'default_integrations.dart';
 import 'hub.dart';
 import 'hub_adapter.dart';
 import 'noop_hub.dart';
+import 'noop_isolate_error_integration.dart'
+    if (dart.library.io) 'isolate_error_integration.dart';
 import 'protocol.dart';
 import 'sentry_client.dart';
 import 'sentry_options.dart';
+import 'utils.dart';
 
 /// Configuration options callback
 typedef OptionsConfiguration = FutureOr<void> Function(SentryOptions);
 
+// run main app callback
+typedef AppRunner = void Function();
```

Comment:
```suggestion
typedef AppRunner = FutureOr<void> Function();
```

just to not be a limitation of the Function signature

Reply:
Why do we need this ? Future<void> and FutureOr<void> should not be used for the signature of a method without `return`

Reply:
if their init methods is async, we must `await` or the SDK won't be fully initialized, I guess?
What I mean is, we'd not force them to just pass a sync method, but could be `async` as well

Reply:
I'm not sure if its the right thing to do, its food for thought

Reply:
you don't have to define a `Future<void>` to use async and await, so it's not a limitation to pass a sync, it force only to pass a method which returns nothin.

Reply:
so if the passed function has an `async` method, it'll work as expected? I needed to make the `OptionsConfiguration` a `FutureOr` for this reason.

Reply:
it's because the optionConfiguration is awaited 
-  https://github.com/getsentry/sentry-dart/blob/1e36bd0e66d0a0b22a00cb954a8d90c050183334/flutter/lib/src/sentry_flutter.dart#L31
- https://github.com/getsentry/sentry-dart/blob/1e36bd0e66d0a0b22a00cb954a8d90c050183334/dart/lib/src/sentry.dart#L29

Reply:
yep, now I ask, does it make sense to `await` the `callback` if we want that callback to be fully run on the `runZonedGuarded`, because that's the way to capture the exceptions in there, I don't know if it'll be caught if an exception happens and there's no `await`

Reply:
✅